src/charm.py

```diff
@@ -105,6 +110,22 @@ def _on_metrics_endpoint_relation_broken(self, event: RelationDepartedEvent):
         username = metrics_username(event.relation)
         self.control_socket.remove_metrics_user(username)
 
+    def _on_dbcluster_relation_changed(self, event):
+        ips = self.model.get_binding(event.relation).network.ingress_addresses
+
+        if len(ips) > 1:
+            logger.error('multiple possible DB bind addresses; set a dbcluster network binding')
+            self.unit.status = BlockedStatus(
+                'multiple possible DB bind addresses; set a dbcluster network binding')
+            return
+
+        ip = str(ips[0])
+        if self._db_bind_address == ip:
+            return
+
+        self._db_bind_address = ip
+        event.relation.data[self.unit].update({'db-bind-address': ip})
```

Comment:
the remote unit can always access a remote's IP via `event.relation.data[event.relation.unit]["ingress-address"], although I'm not sure if it's the same 'ingress address' we're talking about here.

Reply:
try `jhack show-relation --show-juju-keys db controller` to see for yourself what's in there

Reply:
`ingres-address` as set automatically during scope entry is actually just pulled from the zero index of the sorted addresses determined by retrieving network info (the same backing as the call to the `network-get` tool.

We actually want to give the assurance that the list is constituted by a single unique entry, hence the logic here. Going into the blocked state if we don't have a unique address is a message to the user that they need to set the endpoint binding to an appropriate space.